### PR TITLE
Feat/#103 error pages

### PR DESCRIPTION
**Pull request by ciaranschutte**
added:

- pages for routing
- components for ui
- handful of svgs! (thus the large change file count)
- maintenance page toggle

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ const App = ({
 };
 
 App.getInitialProps = async ({ ctx, Component }: AppContext & { Component: PageWithConfig }) => {
-  const pageProps = await Component.getInitialProps({ ...ctx });
+  const pageProps = Component.getInitialProps && (await Component.getInitialProps({ ...ctx }));
```
Comment (ciaranschutte) [inline pages/_app.tsx:45]: some error pages won't have `getInitialProps` func so check before running it

File: components/NavBar.tsx [hunk 2]
```diff
@@ -154,7 +154,7 @@ const LoginButton = () => {
   );
 };
 
-const NavBar = () => {
+const NavBar = ({ hideLinks }: { hideLinks?: boolean }) => {
```
Comment (ciaranschutte) [inline components/NavBar.tsx:157]: hide nav links on error pages

**Comment by ciaranschutte** [comment]
added: 
- pages for routing
- components for ui
- handful of svgs! (thus the large change file count)

File: global/config/index.tsx [hunk 3]
```diff
@@ -10,19 +10,20 @@ export const getConfig = () => {
       `-----BEGIN PUBLIC KEY-----\r\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA0lOqMuPLCVusc6szklNXQL1FHhSkEgR7An+8BllBqTsRHM4bRYosseGFCbYPn8r8FsWuMDtxp0CwTyMQR2PCbJ740DdpbE1KC6jAfZxqcBete7gP0tooJtbvnA6X4vNpG4ukhtUoN9DzNOO0eqMU0Rgyy5HjERdYEWkwTNB30i9I+nHFOSj4MGLBSxNlnuo3keeomCRgtimCx+L/K3HNo0QHTG1J7RzLVAchfQT0lu3pUJ8kB+UM6/6NG+fVyysJyRZ9gadsr4gvHHckw8oUBp2tHvqBEkEdY+rt1Mf5jppt7JUV7HAPLB/qR5jhALY2FX/8MN+lPLmb/nLQQichVQIDAQAB\r\n-----END PUBLIC KEY-----`,
     NEXT_PUBLIC_DAC_API_ROOT: publicConfig.NEXT_PUBLIC_DAC_API_ROOT,
     NEXT_PUBLIC_ARGO_ROOT: publicConfig.NEXT_PUBLIC_ARGO_ROOT || 'https://www.icgc-argo.org/',
-    NEXT_PUBLIC_ARGO_DOCS_ROOT:
-      publicConfig.NEXT_PUBLIC_ARGO_DOCS_ROOT || 'https://docs.icgc-argo.org/',
+    ARGO_DOCS_ROOT: publicConfig.NEXT_PUBLIC_ARGO_DOCS_ROOT || 'https://docs.icgc-argo.org/',
```
Comment (anncatton) [inline global/config/index.tsx:13]: Can we keep the `NEXT_PUBLIC_` prefix for consistency on env vars that are accessed in the ui code?

File: next.config.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ module.exports = withPlugins([withTranspileModules, [withImages], [withFonts]],
     NEXT_PUBLIC_ARGO_ROOT: process.env.NEXT_PUBLIC_ARGO_ROOT,
     NEXT_PUBLIC_ARGO_DOCS_ROOT: process.env.NEXT_PUBLIC_ARGO_DOCS_ROOT,
     NEXT_PUBLIC_ARGO_PLATFORM_ROOT: process.env.NEXT_PUBLIC_ARGO_PLATFORM_ROOT,
+    MAINTENANCE_MODE_ON: process.env.NEXT_PUBLIC_MAINTENANCE_MODE_ON,
```
Comment (anncatton) [inline next.config.js:17]: same as above

**Review by anncatton — APPROVED** [review]
just one small comment!

**Pull request merged**